packages/algob/src/lib/dryrun.ts

```diff
@@ -0,0 +1,153 @@
+  private async createDryRunReq (): Promise<modelsv2.DryrunRequest> {
+    let [_, signedTxn] = await makeAndSignTx(this.deployer, this.execParams, new Map());
+    if (!Array.isArray(signedTxn)) { signedTxn = [signedTxn]; }
+
+    const encodedTxns = signedTxn.map(tx => decodeSignedTransaction(tx));
+    return new modelsv2.DryrunRequest({
+      txns: encodedTxns,
+      sources: undefined
+    });
+  }
+
+  /**
+   * Gets dryrun response in json from the request object
+   * Returns a response with disassembly, logic-sig-messages with PASS/REJECT and sig-trace
+   * @param outFile name of file to dump the response. Dumped in `assets/<file>`
+   * @param force if true, overwrites an existing dryrun response dump
+   */
+  async dryRunResponse (outFile?: string, force?: boolean): Promise<Object> {
+    const dryRunRequest = await this.createDryRunReq();
+    const dryRunResponse = await this.deployer.algodClient.dryrun(dryRunRequest).do();
+    if (outFile) {
+      const outPath = path.join(ASSETS_DIR, outFile);
+      await writeToFile(JSON.stringify(dryRunResponse, null, 2), force ?? false, outPath);
+    }
+    return dryRunResponse;
+  }
+
+  /**
+   * Start a debugger session using child_process.spawn() using the given args.
+   * Kills an existing session first (using killall -9 <process>)
+   * @param tealdbgArgs arguments to `tealdbg debug`. Eg. `--mode signature`, `--group-index 0`
+   */
+  protected async runDebugger (tealdbgArgs: string[]): Promise<boolean> {
+    spawn(`killall`, ['-9', 'tealdbg']); // kill existing tealdbg process first
+
+    const childProcess = spawn(`tealdbg`, ['debug', ...tealdbgArgs], {
+      stdio: "inherit",
+      cwd: process.cwd()
+    });
+    return await new Promise<boolean>((resolve, reject) => {
+      childProcess.once("close", (status) => {
+        childProcess.removeAllListeners("error");
+
+        if (status === 0) {
+          resolve(true);
+          return;
+        }
+
+        reject(new Error("script process returned non 0 status"));
+      });
+
+      childProcess.once("error", (status) => {
+        childProcess.removeAllListeners("close");
+        reject(new Error("script process returned non 0 status"));
+      });
+    });
+  }
+
+  /**
+   * Sets args to pass to `tealdbg debug` command. Currently supported args are
+   * tealFile, mode, groupIndex.
+   * @param debugCtxParams args passed by user for debugger session
+   * @param pathToCache path to --dryrun-dump (msgpack encoded) present in `/cache/dryrun`
+   */
+  private getTealDbgParams (debugCtxParams: DebuggerContext, pathToCache: string): string[] {
+    const tealdbgArgs = [];
+
+    /* Push path of tealfile to debug. If not passed then debugger will use assembled code by default
+     * Supplying the program will allow debugging the "original source" and not the decompiled version. */
+    if (debugCtxParams.tealFile) {
+      const pathToFile = getPathFromDirRecursive(ASSETS_DIR, debugCtxParams.tealFile) as string;
```

Comment:
yes we need `as string`. It's a Typescript compiler issue as it can also return `undefined` if a warning message is passed. Code will not break (as in this case error will be thrown if file is not found).